rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,677 @@
+
+The assumption that we can select/project parts of a subtype that make up a supertype is essential for keeping strictness in JSON Schemas.
+
+And unfortunately, specifying `{ "unevaluatedProperties": false }` does not behave as expected when composing types together in JSON Schema either, which means we will have to redefine how `{ "additionalProperties": true }` or `{ "unevaluatedProperties": false }` behaves within our type extension system to make supertypes keep strictness while allowing composition.
+
+Concrete examples of how JSON Schema breaks with these validation constraints are shown in the [Reference-level explanation](#additional-properties-on-types-1)
+
+## Defining extended types
+
+Extended types will be defined with conventional JSON Schema syntax, the `allOf` keyword. When creating a new entity type it's possible to extend another entity type by adding an entry to `allOf` value with a versioned URI reference.
+Using a versioned URI makes it so that subtypes aren't automatically updated when the supertype is.
+
+As extended types can extend other extended types, we must also make sure that there are no cycles within the type hierarchy, as it could lead to hard to reason about types and unpredictability.
+
+## Addressing previous considerations
+
+In the [Type System RFC](./0352-graph-type-system.md#interfacing-with-types-1), we made the following statement:
+
+> The update messages of both the current and new systems make use of partial schemas, merging the schema given in the message contents with the existing Entity Type. This may or may not be the desired semantics of updating, and could lead to undesired behavior. In that case, the semantics can be changed to treat updates as a complete replacement. This is to be decided and can be considered out of scope for this RFC as it touches on inheritance/forking concepts.
+
+As the new Block Protocol type system doesn't require a delta-based storage approach for its schemas, it's unclear what the advantages of partial schemas for update requests are. We propose that we should treat type updates as complete replacements, so implementation is much more straightforward for embedding applications. Partial schema updates also add some level of indirection, and that may obfuscate error sources and error reasons. Therefore, updates to types must be complete replacements rather than partial schema updates.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## Defining extended types
```

Comment:
The examples, and explanation from what I've seen above, does not Entity Types constraining the existence of links. I think we _should_ talk about that at this point because the concept of `"ordered"`, `"type": "array"`, and the link base URI is something we can decide on compatibility with at the moment.